src/coreComponents/linearAlgebra/interfaces/hypre/HyprePreconditioner.cpp

```diff
@@ -36,55 +38,55 @@ HyprePreconditioner::HyprePreconditioner( LinearSolverParameters params,
   : Base{},
   m_parameters( std::move( params ) ),
   m_precond{},
-  m_functions( std::make_unique< HyprePrecFuncs >() )
+  m_functions( std::make_unique< HyprePrecFuncs >() ),
+  m_dofManager( dofManager ),
+  m_nearNullKernel( nullptr ),
+  m_nullSpacePointer{}
 {
+  // Basic setup for functions
+  if( !m_functions )
+  {
+    m_functions = std::make_unique< HyprePrecFuncs >();
+  }
+
+  // Basic setup common for all preconditioners
   if( m_precond == nullptr )
   {
-    switch( m_parameters.preconditionerType )
-    {
-      case LinearSolverParameters::PreconditionerType::none:
-      {
-        m_functions->setup = (HYPRE_PtrToParSolverFcn) hypre_ParKrylovIdentitySetup;
-        m_functions->apply = (HYPRE_PtrToParSolverFcn) hypre_ParKrylovIdentity;
-        break;
-      }
-      case LinearSolverParameters::PreconditionerType::jacobi:
-      {
-        m_functions->setup = (HYPRE_PtrToParSolverFcn) HYPRE_ParCSRDiagScaleSetup;
-        m_functions->apply = (HYPRE_PtrToParSolverFcn) HYPRE_ParCSRDiagScale;
-        break;
-      }
-      case LinearSolverParameters::PreconditionerType::amg:
-      {
-        createAMG();
-        break;
-      }
-      case LinearSolverParameters::PreconditionerType::mgr:
-      {
-        createMGR( dofManager );
-        break;
-      }
-      case LinearSolverParameters::PreconditionerType::iluk:
-      {
-        createILU();
-        break;
-      }
-      case LinearSolverParameters::PreconditionerType::ilut:
-      {
-        createILUT();
-        break;
-      }
-      default:
-      {
-        GEOSX_ERROR( "Preconditioner type not supported in hypre interface: " << m_parameters.preconditionerType );
-      }
-    }
+    createHyprePreconditioner( m_dofManager );
+  }
+  m_ready = true;
+}
+
+HyprePreconditioner::HyprePreconditioner( LinearSolverParameters params,
+                                          array1d< HypreVector > const & nearNullKernel,
+                                          DofManager const * const dofManager )
+  : Base{},
+  m_parameters( std::move( params ) ),
+  m_precond{},
+  m_functions( std::make_unique< HyprePrecFuncs >() ),
+  m_dofManager( dofManager ),
+  m_nearNullKernel( &nearNullKernel ),
+  m_nullSpacePointer{}
+{
+  // Basic setup for functions
+  if( !m_functions )
+  {
+    m_functions = std::make_unique< HyprePrecFuncs >();
+  }
+
+  // Basic setup common for all preconditioners
+  if( m_precond == nullptr )
+  {
+    createHyprePreconditioner( m_dofManager );
```

Comment:
Same comment as in the other constructor

Reply:
Ok.